sdb/command.py

```diff
@@ -34,22 +34,68 @@ class Command:
 
     # pylint: disable=too-few-public-methods
 
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        #
+        # If the class doesn't have a docstring, "inspect.getdoc" will
+        # pull from the parent class's docstring, if the parent class
+        # does have a docstring. This is not the behavior we want, so we
+        # first check "cls.__doc__" before using "inspect.getdoc".
+        #
+        if cls.__doc__:
+            summary = inspect.getdoc(cls).splitlines()[0].strip()
+        else:
+            summary = None
+        return argparse.ArgumentParser(prog=name, description=summary)
+
+    @classmethod
+    def help(cls, name: str):
+        """
+        This method will print a "help message" for the particular
+        command class that it's called on. The docstring and parser for
+        the class is used to populate the contents of the message.
+        """
+        parser = cls._init_parser(name)
+
+        print("SUMMARY")
+        for i, line in enumerate(parser.format_help().split('\n')):
+            #
+            # When printing the help message, the first line will have a
+            # "usage: " prefix string which looks awkward, so we strip
+            # that prefix prior to printing the first line.
+            #
+            if i == 0:
+                line = line.replace('usage: ', '')
+            print("    {}".format(line.replace('usage: ', '')))
+
+        if len(cls.names) > 1:
+            print("ALIASES")
+            print("    {}".format(", ".join(cls.names)))
+            print()
+
+        #
+        # If the class doesn't have a docstring, "inspect.getdoc" will
+        # pull from the parent class's docstring, if the parent class
+        # does have a docstring. This is not the behavior we want, so we
+        # first check "cls.__doc__" before using "inspect.getdoc".
+        #
+        if cls.__doc__:
+            #
+            # The first line of the docstring is the summary, whichis
+            # already be included in the parser description. The second
+            # line should be empty. Thus, we skip these two lines.
+            #
+            for line in inspect.getdoc(cls).splitlines()[2:]:
+                print("{}".format(line))
+            print()
+
     #
     # names:
     #    The potential names that can be used to invoke
     #    the command.
     #
     names: List[str] = []
 
-    #
-    # name:
-    #    The name used when the command was invoked. This
-    #    is generally used as a parameter passed to
-    #    exceptions raised by SDB to make error messages
-    #    more precise.
-    #
-    name: str = ""
-
```

Comment:
Out of curiosity, why is this removed? Asking because I liked the comment making the differentiation between `name` and `names`.

Reply:
So, based on my understanding of python, this will create a class level variable named `name`, which is different than the object level variable named `name`. I didn't see where we used the class level `name` variable, vs. how we do use the object level `self.name` variable.

Assuming I'm correct and this is dead code, I figured it should be removed to avoid confusion.

Let me know if I'm understanding this incorrectly.

Reply:
I see. I honestly had no idea that this was a thing. I'm resolving this.